templates/viewer.php

```diff
@@ -282,13 +283,13 @@
                   <span data-l10n-id="print_label">Print</span>
                 </button>
 
-                <button id="download" class="toolbarButton hiddenMediumView" title="Save" tabindex="33" data-l10n-id="save" hidden="true">
+                <button id="download" class="toolbarButton hiddenMediumView" title="Save" tabindex="33" data-l10n-id="save">
                   <span data-l10n-id="save_label">Save</span>
                 </button>
 
                 <div class="verticalToolbarSeparator hiddenMediumView"></div>
 
-                <div id="editorModeButtons" class="splitToolbarButton toggled" role="radiogroup" hidden="true">
+                <div id="editorModeButtons" class="splitToolbarButton toggled" role="radiogroup">
```

Comment:
Annotation and save buttons should not be visible if the file can not be edited (for example, if it is shared with another user as read-only). This could be done by keeping the buttons hidden by default and only showing them if the file permissions are checked and it is editable.

The PDFView Vue component receives the properties from the [Mime mixin of the Viewer](https://github.com/nextcloud/viewer/blob/fa209648d0cb6deb292bb859c7f151d7def47d91/src/mixins/Mime.js), which is populated with the file information (including permissions) [when the viewer is opened](https://github.com/nextcloud/viewer/blob/e1f2c85b4dffae974845ca57ceb19055ce26a6a3/src/views/Viewer.vue#L611-L614). Therefore, when the PDFView component is mounted, the permissions could be checked and the buttons enabled once the iframe was loaded.

Note that the PDFView component was not used in public pages for a single PDF file, but this was fixed in #811 so the check can be easily implemented here for all cases :-)